src/ar_gateway_middleware.erl

```diff
@@ -132,21 +139,29 @@ handle_unknown_request(Req, Env) ->
 
 resolve_path(Path) ->
 	case ar_http_iface_server:split_path(Path) of
-		[] -> root;
-		[<<Hash:43/bytes>>] -> resolve_path_1(Hash);
-		[_ | _] -> other
+		[] ->
+			root;
+		[<<Hash:43/bytes>> | SubPathSegments] ->
+			resolve_path_1(Hash, SubPathSegments);
+		[_ | _] = PathSegments ->
+			OtherPath = list_to_binary(lists:join(<<"/">>, PathSegments)),
+			{other, OtherPath}
 	end.
 
-resolve_path_1(Hash) ->
+resolve_path_1(Hash, SubPathSegments) ->
 	case get_tx_from_hash(Hash) of
-		{ok, TXID, Filename} -> {tx, TXID, Filename};
-		invalid -> other;
-		not_found -> not_found
+		{ok, TXID, Filename} ->
+			SubPath = list_to_binary(lists:join(<<"/">>, SubPathSegments)),
+			{tx, TXID, Filename, SubPath};
+		invalid ->
+			other;
```

Comment:
The callers seem to always expect a `{other, _}` tuple now.

Reply:
Good catch 👍  Fixed in eb5b42ee